src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/EmptyBlockCheck.java

```diff
@@ -195,8 +195,10 @@ protected boolean hasText(final DetailAST slistAST) {
         }
         else {
             // check only whitespace of first & last lines
-            if (lines[slistLineNo - 1].substring(slistColNo + 1).trim().isEmpty()
-                    && lines[rcurlyLineNo - 1].substring(0, rcurlyColNo).trim().isEmpty()) {
+            if (CommonUtils.isOnlyWhitespacesOrEmpty(lines[slistLineNo - 1]
+                    .substring(slistColNo + 1))
+                    && CommonUtils.isOnlyWhitespacesOrEmpty(lines[rcurlyLineNo - 1]
+                    .substring(0, rcurlyColNo))) {
                 // check if all lines are also only whitespace
```

Comment:
This become a messy, please move first and last to local variables, comment above should be removed as it completely moved to code variable and method names.